packages/keystone/crypto/EncryptionManager.spec.js

```diff
@@ -0,0 +1,202 @@
+const crypto = require('crypto')
+
+const { faker } = require('@faker-js/faker')
+const { groupBy } = require('lodash')
+
+const { EncryptionManager, SUPPORTED_MODES } = require('@open-condo/keystone/crypto/EncryptionManager')
+const { catchErrorFrom } = require('@open-condo/keystone/test.utils')
+
+function generateVersions () {
+    const cipherAlgorithms = crypto.getCiphers()
+    const cipherInfos = cipherAlgorithms.map(alg => crypto.getCipherInfo(alg))
+    const modes = new Set(cipherInfos.map(info => info.mode))
+
+    const secrets = cipherInfos.map((info) => faker.random.alphaNumeric(info.keyLength || 0))
+    const versions = cipherAlgorithms.map((algorithm, i) => ({
+        mode: cipherInfos[i].mode,
+        id: faker.random.alphaNumeric(10),
+        algorithm: algorithm,
+        secret: secrets[i],
+    }))
+    return { versions: groupBy(versions, 'mode'), modes: [...modes] }
+}
+
+function generateVersionsInMode (mode, count = 1) {
+    const cipherAlgorithms = crypto.getCiphers()
+    const cipherInfos = cipherAlgorithms
+        .map(alg => crypto.getCipherInfo(alg))
+        .filter(info => info.mode === mode)
+    const infosForVersions = faker.helpers.arrayElements(cipherInfos, count)
+    const versionsArray = infosForVersions.map(info => {
+        return {
+            id: faker.random.alphaNumeric(10),
+            algorithm: info.name,
+            secret: faker.random.alphaNumeric(info.keyLength || 0),
+        }
+    })
+    return versionsArray.reduce((versions, currentVersion) => {
+        versions[currentVersion.id] = { algorithm: currentVersion.algorithm, secret: currentVersion.secret }
+        return versions
+    }, {})
+}
+
+describe('EncryptionManager', () => {
+    const { versions: versionsByMode, modes } = generateVersions()
+    const unsupportedModes = modes.filter(mode => !SUPPORTED_MODES.includes(mode))
+
+    describe('Supported algorithm modes', () => {
+
+        describe.each(SUPPORTED_MODES)('%p', (mode) => {
+            const versions = versionsByMode[mode]
+
+            test.each(versions)('$algorithm', (version) => {
+                const manager = new EncryptionManager({ versions: { [version.id]: version }, encryptionVersionId: version.id })
+
+                const initialString = faker.random.alphaNumeric(20)
+                const encrypted = manager.encrypt(initialString)
+                expect(encrypted).not.toEqual(initialString)
+                expect(encrypted.split(':')).toHaveLength(4)
```

Comment:
Really it's better to use pattern like `<name>:<base64-encoded-data>` and you don't need to validate anything